portal-kernel/src/com/liferay/portal/kernel/security/auth/FullNameGeneratorFactory.java

```diff
@@ -27,6 +27,6 @@ private FullNameGeneratorFactory() {
 	private static volatile FullNameGenerator _fullNameGenerator =
 		ServiceProxyFactory.newServiceTrackedInstance(
 			FullNameGenerator.class, FullNameGeneratorFactory.class,
-			"_fullNameGenerator", true);
+			"_fullNameGenerator", false);
```

Comment:
@JanisZhang If the `FullNameGenerator` is instantiated by Spring, could you just change this inject the instance by Spring, and rename the class to `FullNameGeneratorUtil`?

Reply:
Hi @dantewang, `FullNameGenerator` has two subclasses named `DefaultFullNameGenerator` and `FamilyNameFirstFullNameGenerator`. the `DefaultFullNameGenerator` is the only one injected using Spring configuration. I am not sure whether the FamilyNameFirstFullNameGenerator will be invoked if we change it in this way.

Reply:
@JanisZhang Which one is actually in that field, the `Default` or the `FamilyNameFirst` one?

Reply:
@dantewang  I have done some tests, the actual value in the field is the default one.